reports-backend/src/main/java/org/jboss/da/reports/impl/ReportsGeneratorImpl.java

```diff
@@ -369,4 +549,66 @@ private ProductArtifact toProductArtifact(ProductVersionArtifactRelationship pa)
         return ret;
     }
 
+    @Override
+    public List<LookupReport> getLookupReportsForGavs(LookupGAVsRequest request) {
+        
+        // Use a concurrent list since we're using parallel stream to add stuff to the list
+        CopyOnWriteArrayList<LookupReport> reportsList = new CopyOnWriteArrayList<>();
```

Comment:
I know that you just took it from elsewhere, but using concurrent set might be better then resource heavy CopyOnWriteArrayList. Eg ``ConcurrentHashMap.newKeySet()``